src/bridges/tokemak/TokemakBridge.sol

```diff
@@ -0,0 +1,353 @@
+    }
+
+    /**
+     * @dev Function to finalise an interaction
+     * Converts the held amount of tranche asset for the given interaction into the output asset
+     * @param _interactionNonce The _nonce value for the interaction that should be finalised
+     * @return outputValueA The interaction's first ouptut value after this call
+     * @return outputValueB The interaction's second ouptut value after this call - will be 0
+     * @return interactionCompleted Flag specifying if this interaction is completed
+     */
+    function finalise(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint256 _interactionNonce,
+        uint64
+    )
+        external
+        payable
+        override(BridgeBase)
+        returns (
+            uint256 outputValueA,
+            uint256 outputValueB,
+            bool interactionCompleted
+        )
+    {
+        if (msg.sender != ROLLUP_PROCESSOR) revert ErrorLib.InvalidCaller();
+        if (_inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidInputA();
+        if (_outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidOutputA();
+
+        address _tAsset = _inputAssetA.erc20Address;
+        if (_tAsset == address(0)) revert ErrorLib.InvalidInputA();
+
+        // Pending withdrawal value
+        uint256 _inputValue = pendingInteractions[_interactionNonce].inputValue;
+        if (_inputValue == 0) revert ErrorLib.InvalidInput();
+
+        // Withdraw pending withdrawals
+        (outputValueA, interactionCompleted) = _finaliseWithdraw(_tAsset, _inputValue, _interactionNonce);
+    }
+
+    /**
+     * @dev Function called to automatically fetch all the Pools from Tokemak Manager
+     */
+    function loadTokens() public {
+        IManager _manager = IManager(MANAGER);
+        address[] memory _pools = _manager.getPools();
+        for (uint256 i = 0; i < _pools.length; i++) {
+            address _pool = _pools[i];
+            Ttoken _token = Ttoken(_pool);
+
+            address _asset = _token.underlyer();
+            IERC20 _assetToken = IERC20(_asset);
+
+            _assetToken.approve(ROLLUP_PROCESSOR, MAX_UINT);
+            _assetToken.approve(_pool, MAX_UINT);
+            _token.approve(ROLLUP_PROCESSOR, MAX_UINT);
+            _token.approve(_pool, MAX_UINT);
+
+            tTokens[_asset] = _pool;
+            assets[_pool] = _asset;
+        }
+    }
+
+    /**
+     * @dev Function to attempt finalising of as many interactions as possible within the specified gas limit
+     * Continue checking for and finalising interactions until we expend the available gas
+     * @param _gasFloor The amount of gas that needs to remain after this call has completed
+     */
+    function _finalisePendingInteractions(uint256 _gasFloor) internal {
```

Comment:
`_gasFloor` param is always set to `MIN_GAS_FOR_FUNCTION_COMPLETION` so it doesn't make much sense to having it passed.